src/Makefile.am

```diff
@@ -1,7 +1,7 @@
 ## Process this file with autoconf to produce configure.
 
-AM_CFLAGS   = -Wall -Wwrite-strings $(PTHREAD_CFLAGS)
-AM_CXXFLAGS = -Wall -Wwrite-strings -Woverloaded-virtual $(PTHREAD_CFLAGS)
+AM_CFLAGS   = -Wall -Wwrite-strings $(PTHREAD_CFLAGS) -fgnu-tm
```

Comment:
gcc 4.7 introduces -fgnu-tm. How about checking the compiler version before enabling MPMC code?